sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -260,18 +261,127 @@ public static final Table from(ColumnarBatch batch) {
     return new Table(extractBases(batch));
   }
 
+  /**
+   * Get the data types for a batch.
+   */
+  public static final List<DataType> extractTypes(ColumnarBatch batch) {
```

Comment:
nit: I don't like List.  I think it would be better to use an array because it integrates better with scala, but the API in cudf is for a List, so we might want a follow on fix there too.